### PR TITLE
Fix #5903: New attribute in Domain for Solana Dapps Permission Management 

### LINKED ISSUE #5903: Wallet: Add a new attribute in Domain to track Solana Dapps permissions with user accounts

### PR DESCRIPTION
**Pull request by nuo-xu**
## Summary of Changes
Added a new attribute `wallet_solanaPermittedAcccounts` which has the same logic to store permitted solana account public key. 
Update apis to take coin type to avoid code duplication. 
Add new and update related unit tests.

This pull request fixes #5903

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [ ] User-facing strings use `NSLocalizableString()`
- [ ] New or updated UI has been tested across:
  - [ ] Light & dark mode
  - [ ] Different size classes (iPhone, landscape, iPad)
  - [ ] Different dynamic type sizes

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/Settings/Clearables.swift [hunk 1]
```diff
@@ -54,7 +54,7 @@ class CookiesAndCacheClearable: Clearable {
     UserDefaults.standard.synchronize()
     await BraveWebView.sharedNonPersistentStore().removeData(ofTypes: WKWebsiteDataStore.allWebsiteDataTypes(), modifiedSince: Date(timeIntervalSinceReferenceDate: 0))
     UserDefaults.standard.synchronize()
-    await Domain.clearAllEthereumPermissions()
+    await Domain.clearAllWalletPermissions(for: .eth)
```
Comment (StephenHeaps) [inline Client/Frontend/Settings/Clearables.swift:57]: I think we need to clear permissions for Solana coin type here too

File: Sources/BraveWallet/Crypto/Stores/KeyringStore.swift [hunk 2]
```diff
@@ -241,7 +241,7 @@ public class KeyringStore: ObservableObject {
       }
       self.walletService.setSelectedCoin(.eth)
       self.rpcService.setNetwork(BraveWallet.MainnetChainId, coin: .eth, completion: { _ in })
-      Domain.clearAllEthereumPermissions()
+      Domain.clearAllWalletPermissions(for: .eth)
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Stores/KeyringStore.swift:244]: I think we need to clear permissions for Solana coin type here too as well

File: Sources/BraveWallet/Crypto/Stores/SettingsStore.swift [hunk 3]
```diff
@@ -64,7 +64,7 @@ public class SettingsStore: ObservableObject {
   func reset() {
     walletService.reset()
     keychain.resetPasswordInKeychain(key: KeyringStore.passwordKeychainKey)
-    Domain.clearAllEthereumPermissions()
+    Domain.clearAllWalletPermissions(for: .eth)
```
Comment (StephenHeaps) [inline Sources/BraveWallet/Crypto/Stores/SettingsStore.swift:67]: And here

File: Sources/Data/models/Domain.swift [hunk 4]
```diff
@@ -154,41 +157,91 @@ public final class Domain: NSManagedObject, CRUD {
...
+          case .sol:
+            $0.wallet_solanaPermittedAcccounts = nil
+          case .fil:
+            break
+          @unknown default:
+            break
+          }
         }
       } catch {
         log.error("Clear ethereum permissions error: \(error)")
```
Comment (StephenHeaps) [inline Sources/Data/models/Domain.swift]: I think should update this error log to include Solana while we're here

**Review by StephenHeaps — CHANGES_REQUESTED** [review]

File: Sources/Data/models/Model.xcdatamodeld/.xccurrentversion [hunk 5]
```diff
@@ -3,6 +3,6 @@
 <plist version="1.0">
 <dict>
 	<key>_XCCurrentVersionName</key>
-	<string>Model15.xcdatamodel</string>
+	<string>Model16.xcdatamodel</string>
```
Comment (iccub) [inline Sources/Data/models/Model.xcdatamodeld/.xccurrentversion]: I do not think migration is needed when adding a new attribute. Or do you update existing ones too?

**Review by iccub — CHANGES_REQUESTED** [review]

File: Sources/Data/models/Model.xcdatamodeld/.xccurrentversion (hunk 5 above)
Comment (nuo-xu) [reply Sources/Data/models/Model.xcdatamodeld/.xccurrentversion]: Good catch. Was debating changing the existed one at the beginning but ended up with no change. Will revert this change.

File: Client/Frontend/Settings/Clearables.swift (hunk 1 above)
Comment (nuo-xu) [reply Client/Frontend/Settings/Clearables.swift:57]: will integrate clearing solana permission once we support solana dapps

File: Sources/BraveWallet/Crypto/Stores/KeyringStore.swift (hunk 2 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Stores/KeyringStore.swift:244]: will integrate clearing solana permission once we support solana dapps

File: Sources/BraveWallet/Crypto/Stores/SettingsStore.swift (hunk 3 above)
Comment (nuo-xu) [reply Sources/BraveWallet/Crypto/Stores/SettingsStore.swift:67]: will integrate clearing solana permission once we support solana dapps

**Review by iccub — APPROVED** [review]

**Review by StephenHeaps — APPROVED** [review]

File: Sources/Data/models/Model.xcdatamodeld/.xccurrentversion (hunk 5 above)
Comment (nuo-xu) [reply Sources/Data/models/Model.xcdatamodeld/.xccurrentversion]: resolve this comment as its been fixed in [c5f423c](https://github.com/brave/brave-ios/pull/5910/commits/c5f423ca50c586bafae77e71e382a480eb2d05e5)

**Pull request merged**